amethyst_animation/src/skinning/systems.rs

```diff
@@ -40,11 +40,19 @@ impl<'a> System<'a> for VertexSkinningSystem {
 
     fn run(&mut self, (joints, global_transforms, mut skins, mut matrices): Self::SystemData) {
         self.updated.clear();
-        global_transforms
-            .populate_modified(&mut self.updated_id.as_mut().unwrap(), &mut self.updated);
-        global_transforms
-            .populate_inserted(&mut self.inserted_id.as_mut().unwrap(), &mut self.updated);
+
+        match self.updated_id.as_mut() {
+            Some(x) => { global_transforms.populate_modified(x, &mut self.updated) },
+            None => { warn!("VertexSkinningSystem missing updated_id.") }, 
```

Comment:
Is there a realistic scenario where this would happen and not be a bug?

Reply:
Only if Setup() is not ran or if the value is modified to None but I don't think it hurts, I guess.

Reply:
I would assume of course that Setup() is ran for every system before it's Run()

Reply:
Found the diff, I guess this is expected. https://github.com/amethyst/amethyst/commit/d949f6c603bb8c3efbc3e12ee93d26fd2b1ff734#diff-91175d8ffab11be5a6cef3d04164cf82R48

Fixed.